conans/test/functional/toolchains/cmake/test_cmake.py

```diff
@@ -36,10 +36,10 @@ def generate(self):
                 tc.variables.release["MYVAR_CONFIG"] = "MYVAR_RELEASE"
                 tc.variables.debug["MYVAR2_CONFIG"] = "MYVAR2_DEBUG"
                 tc.variables.release["MYVAR2_CONFIG"] = "MYVAR2_RELEASE"
-                tc.preprocessor_definitions["MYDEFINE"] = "MYDEF_VALUE"
+                tc.preprocessor_definitions["MYDEFINE"] = "\"MYDEF_VALUE\""
```

Comment:
I am not sure about this, that means that I, as a user, always need to double quote strings, or they will fail? That will surely bring some headaches to users that don't do that in the first place. Even if it is in the docs, it is very easy to omit the double quotes. I think that as long as the input is a string, the quotes should happen in the Conan internals as needed.

Reply:
But how can we know that the value must be quoted? Can we distinguish between these two?

```
#define MYDEFINE MYDEF_VALUE
#define MYDEFINE "MYDEF_VALUE_STR"
```
I thin that you should care about what you want to define, and you could also use: 

```
tc.preprocessor_definitions["MYDEFINE"] = '"MYDEF_VALUE"'
```

Reply:
I see.
Then I guess you could do:
```python
tc.preprocessor_definitions["MYDEFINEINT"] = "42"
```
And the result will be identical, mapping to an integer in the C++ definition?

Reply:
> I see. Then I guess you could do:
> 
> ```python
> tc.preprocessor_definitions["MYDEFINEINT"] = "42"
> ```
> 
> And the result will be identical, mapping to an integer in the C++ definition?

Yes, that should define an integer in that case

Reply:
How do I then define a string with only numerical contents? Like "12345" ?

Reply:
You could do 

`tc.preprocessor_definitions["MYDEFINEINT"] = 42`

that would be equivalent to:

`tc.preprocessor_definitions["MYDEFINEINT"] = "42"`

I don't think we should try from Conan to interpret if the definition is an int or not, because you could want to define something like this:

`tc.preprocessor_definitions["OTHERINT"] = "MYDEFINEINT"`

and you want that to be declared like `#define OTHERINT MYDEFINEINT` and not like `#define OTHERINT "MYDEFINEINT"`